includes/class-wcpdf-endpoint.php

```diff
@@ -76,7 +76,12 @@ public function get_document_link( $order, $document_type, $additional_vars = ar
 		if ( is_user_logged_in() ) {
 			$access_key = wp_create_nonce( $this->action );
 		} elseif ( ! is_user_logged_in() && isset( $debug_settings['guest_access'] ) ) {
```

Comment:
@jhosagid7 I think you only need to filter `$debug_settings['guest_access']` here. The rest of the new code within this function, including the new parameter, has no sense to me.